File: be/src/main/java/jointscamp/be/entity/Produk.java
package jointscamp.be.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.*;
import lombok.*;
import lombok.extern.jackson.Jacksonized;
import org.hibernate.annotations.*;

import java.util.Date;
import java.util.UUID;

@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@DynamicUpdate
@DynamicInsert
@ToString
public class Produk {

    @Id
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @GeneratedValue(generator = "uuid")
    private UUID id;

    private String namaProduk;

    private String detailProduk;

    @Lob
    private byte[] gambarProduk;

    private String lokasiProduk;

    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @UpdateTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date updatedAt;

    @Temporal(TemporalType.TIMESTAMP)
    private Date deletedAt;

    @ManyToOne(fetch = FetchType.LAZY, targetEntity = User.class)
    @JsonIgnore
    @ToString.Exclude
    private User user;
}

File: jointscamp/src/main/kotlin/centkarts/jointscamp/entity/Produk.kt
package centkarts.jointscamp.entity

import com.fasterxml.jackson.annotation.JsonKey
import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.UpdateTimestamp
import java.util.Date
import java.util.UUID

@Entity
class Produk(

    var productName: String,
    var productDetail: String,
    var productImage: String?,
    var productLocation: String?,

    @ManyToOne(fetch = FetchType.LAZY)
    var user: User,

    @OneToMany(fetch = FetchType.LAZY, cascade = [CascadeType.ALL], mappedBy = "produk")
    var like: List<Like>,

    @CreationTimestamp
    var createdAt: Date,

    @UpdateTimestamp
    var updatedAt: Date,

    var deletedAt: Date?,

    @Id
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @GeneratedValue(generator = "uuid")
    var id: UUID?
){}